File: src/model/questionmodel.ts
import  { Question } from "@prisma/client"

export interface QuestionResponse {
    id: number
    question: string
}

// Fungsi untuk mengonversi Survey menjadi SurveyResponse
export function toQuestionResponseList(prismaQuestion: Question[]): QuestionResponse[] {
    const result = prismaQuestion.map((question) => {
        return {
            id: question.id,
            question: question.question,
        }
    })

    return result
}

export function toQuestionResponse(prismaQuestion: Question): QuestionResponse {
    return {
    id: prismaQuestion.id,
    question: prismaQuestion.question,
    }
}

export interface QuestionCreateRequest {
   question: string
   survey_id: number
}

export interface QuestionUpdateRequest {
   question: string
   survey_id: number
}

File: models/QuestionModel.kt
package com.example.alp_vp.models

data class GetAllQuestionResponse(
    val data: List<QuestionModel>
)

data class GetQuestionResponse(
    val data: QuestionModel
)

data class QuestionModel(
    val id: Int,
    val question: String,
    val surveyId: Int
)

fun toQuestionResponseList(questions: List<QuestionModel>): List<QuestionResponse> {
    return questions.map { question ->
        QuestionResponse(
            id = question.id,
            question = question.question,
            surveyId = question.surveyId
        )
    }
}

fun toQuestionResponse(question: QuestionModel): QuestionResponse {
    return QuestionResponse(
        id = question.id,
        question = question.question,
        surveyId = question.surveyId
    )
}

data class QuestionCreateRequest(
    val question: String,
    val surveyId: Int
)


data class QuestionUpdateRequest(
    val question: String,
    val surveyId: Int
)

data class QuestionResponse(
    val id: Int,
    val question: String,
    val surveyId: Int
)


data class Question(
    val id: Int,
    val question: String,
    val surveyId: Int
)
